policy/feature-flags.md

```diff
@@ -6,63 +6,137 @@ Product and release need to track bugs whose visibility is controlled through a
+- A comment is filed listing the name of the preference
+- The `behind-pref` flag is set to `in-progress`
+- The bug's severity is set to `enhancement`
+- The three implementation bugs should be marked as blocking the `[meta]` bug for the new feature
+
+As the feature is developed and the individual patches implement it land, it's kept off by compiler directives, the pref, or both. As these land, and are not backed out, these bugs can be marked RESOLVED FIXED.
+
+Once the dependent bugs have been resolved, the feature can be enabled in nightly. The lead for the feature–which may be an engineer, a program manager, or a product manager–must notify the Nightly Release Manager before enabling it.
+
+- The `[meta]` bug's status is moved to RESOLVED and resolution to FIXED.
+- The bug's `behind-pref` flag is set to `nightlyNN+2` to indicate it's now available in nightly
+- The `qe-verify` flag is set to ?, requesting QA's attention
+
+Before the feature can graduate to Beta, it must be verified by QA. 
+
+- QA agrees to take on verfication, setting `qe-verify` to `+`
+- The feature is tested on nightly and confirmed to work as specified (implicit here is the feature team's involvement in creating a test plan)
+- QA moves the bug's resolution to VERIFIED
+
+If the feature does not pass testing then QA should file bugs blocking the `[meta]` bug for the feature. QA and the development team must confer and decide if the feature will be disabled in Nightly, or allowed to be kept on while bugs are fixed. This will depend on risk and severity of the bugs found. 
 
-It's decided that this should be controlled behind a preference, `browser.newtabpage.default.tabbycat`:
+If it's decided to disable the feature, then it should be turned off in the nightly build and the `behind-pref` flag set to `off`. The bug's comments should explain why that decision was released. Once the defects have been resolved, then `behind-pref` can be reset to `nightlyNN+2`.
 
-*   A comment is filed listing the name of the preference
-*   The `behind-pref` flag is set to +
-*   The `qe-verify` flag is set to ?, requesting QA's attention
+Once the feature has been verfied by QA then:
 
-![Screenshot of Bugzilla with tracking pane open for editing displaying the behind-pref and qe-verify flags](/public/images/feature-flags-editing-in-bmo.png)
+- The bug for the feature should be moved to RESOLVED VERIFIED status
+- The bug should be enabled in Beta once Release Management approves
+- the `behind-pref` flag is updated to `betaNN+1` 
 
-By default, the preference enabling the feature **should not** be set.
+The feature now *rides the trains* to release, and on merge day, `behind-pref` is set to `nightlyNN`. The bug is then considered completed. 
 
-A patch lands in mozilla-central with the code for the feature controlled by the preference, then:
+When the next ESR is released, the `behind-pref-esr` field should be set to the version where it was relased. 
```

Comment:
Either this needs to be automated or we need to have the ESR flag also in advance so we can set this in advance to reflect that it will release in the next ESR until otherwise changed.